test-debt sweep: 5 pre-existing test failures on development (config / scraper-adapter / enrichment agent)

## Problem

5 unit tests are failing on `development` (full-suite run, ignoring integration). All pre-existing — unrelated to current eval-data-gap work — but they're masking real failures and adding noise to PR-CI runs.

Captured 2026-04-28 against `development` HEAD `0d84084` (chore: document qa-eval --use-http env-var pattern in .env.example #300):

```
FAILED common/tests/test_config_loader.py::test_env_override_emits_warning_once
FAILED ingestion/tests/test_scraper_adapter.py::TestScraperAdapter::test_fixture_fallback_loads_records
FAILED ingestion/tests/test_scraper_adapter.py::TestScraperAdapter::test_fixture_field_mapping
FAILED ingestion/tests/test_scraper_adapter.py::TestScraperAdapter::test_health_check_fixture_mode
FAILED tests/test_enrichment_agent.py::TestEnrichmentAgentBatchRecords::test_process_passes_none_session_when_db_unavailable
```

## Per-test breakdown

### 1. `test_env_override_emits_warning_once` (config loader)

**Asserts:** the `config_env_override_used` warning fires *exactly once* per env-var override (rate-limited via a module-level `_emitted_once` set).

**Failure mode:** passes in isolation, fails in the full suite.

**Root cause:** module-level state leakage. An earlier test in the same Python process triggers the same env-override path, populating `_emitted_once`. By the time this test runs, the warning is already de-duped.

**Fix shape:** add an `autouse` fixture (or `setup_method`) that clears `_emitted_once` between tests. Or use `monkeypatch.setattr(common.config_loader, "_emitted_once", set())`.

**Severity:** low (cosmetic; production rate-limit works correctly).

---

### 2-4. `test_scraper_adapter` — three fixture-mode tests

| Test | Asserts | Actual |
|---|---|---|
| `test_fixture_fallback_loads_records` | `len(records) > 0` | `len([]) > 0` |
| `test_fixture_field_mapping` | `len(records) >= 1` | `len([]) >= 1` |
| `test_health_check_fixture_mode` | `result["reachable"] is True` | `False is True` |

**Captured log on each:** `WARNING ingestion.sources.scraper_adapter:101 scraper_fixture_missing`

**Root cause:** the scraper adapter's fixture file isn't found at the path the test expects. Likely a fixture rename / move from PR #285 (git-lfs migration) or PR #280 (config refactor) that didn't update the adapter's hardcoded fixture-path lookup. The adapter logs `scraper_fixture_missing` and gracefully returns `[]` — a degrade path the tests weren't expecting.

**Fix shape:** locate the missing fixture, restore or update the path. Optional: harden the scraper adapter so missing fixture fails loudly in test mode rather than silently degrading (the same anti-pattern that produced the 469 orphans in #289 — silent skip → invisible failure).

**Severity:** medium (blocks scraper-adapter test suite; production uses JSearch not scraper, so production isn't affected).

---

### 5. `test_process_passes_none_session_when_db_unavailable` (enrichment agent)

**Asserts:** when `check_db_connection()` is mocked to return `False`, `session_scope()` is never called during `agent.process(event)`.

**Failure mode:** `session_scope()` is called once (9-line call trace shows two `s.execute(SELECT...).all()` patterns).

**Root cause:** `EnrichmentAgent._ensure_refs()` at `enrichment/agent.py:499` calls `session_scope()` when `_db_url_configured()` is True. The test only patches `check_db_connection` (which `_ensure_refs` doesn't check) — so when `_ensure_refs` is invoked from inside `run_batch`'s row loop (line 749), it tries to load reference labels via DB, ignoring the `check_db_connection` mock entirely. The test asserts a stronger invariant than the code enforces.

**Fix shape:** either patch `_ensure_refs` (or `_db_url_configured`) too, OR make `_ensure_refs` honor `check_db_connection` for consistency with the rest of the agent.

**Severity:** low (test-assertion mismatch, not a real bug).

---

## Recommendation

Fix these as a chore PR (estimated ~1 hour):

- **#1 + #5** are 5-line test fixes (state reset / mock additions)
- **#2-4** require locating the missing fixture file (10–15 min) + either restoring it or updating the lookup path

Worth doing because:
- Pre-existing failures mask real failures in any future PR-CI run
- The `scraper_fixture_missing` graceful-degrade pattern is the same anti-pattern that produced JIE #289's 469 orphans — fixing it raises the floor of "loud failure" hygiene
- Net diff is small; risk is low; payoff is meaningful

## Owner

Triage assigned to @theGaryLarson; reassign to ingestion-touching pair (Pair A historically) for #2-4 and code-health for #1 + #5 once a window opens.

## Out of scope

Renaming the underlying graceful-degrade pattern to fail-loud across the codebase. Doing that systematically is a separate, larger conversation; this issue is just to clean the 5 known failures.

--- a/enrichment/agent.py
+++ b/enrichment/agent.py
@@ -288,14 +288,13 @@ def _emit_enrichment_degraded(
     reason: str,
     extraction_note: str | None,
 ) -> None:
-    message = "Spam classification degraded; record continued with null spam fields."
     log.warning(
         "EnrichmentDegraded",
         posting_id=posting_id,
         normalized_job_id=normalized_job_id,
         reason=reason,
         extraction_note=extraction_note,
-        message=message,
+        detail="Spam classification degraded; record continued with null spam fields.",
     )
     if _alert_bus is None:
         return
@@ -372,7 +371,7 @@ def _check_soc_unclassified_rate(
         unclassified_count=unclassified_count,
         unclassified_rate=round(unclassified_rate, 4),
         threshold=threshold,
-        message=(
+        detail=(
             f"SOC unclassified rate {unclassified_rate:.1%} exceeds threshold "
             f"{threshold:.1%} for batch {batch_id}. "
             "Check dbo.socc population and LLM classification quality."

--- a/enrichment/agent.py
+++ b/enrichment/agent.py
@@ -316,7 +316,7 @@ def _emit_enrichment_degraded(
                     "overall_confidence",
                 ],
                 "extraction_note": extraction_note,
-                "message": message,
+                "message": "Spam classification degraded; record continued with null spam fields.",
             },
         )
         _alert_bus.publish(event)

--- a/ingestion/sources/scraper_adapter.py
+++ b/ingestion/sources/scraper_adapter.py
@@ -18,7 +18,7 @@ from ingestion.sources.base_adapter import SourceAdapter
 log = structlog.get_logger()
 
 _FALLBACK_SCRAPE = (
-    Path(__file__).parent.parent  # agents/
+    Path(__file__).parent.parent.parent  # repo root
     / "data"
     / "fixtures"
     / "fallback_scrape_sample.json"

